code/modules/projectiles/ammo_datums.dm

```diff
@@ -2468,10 +2468,96 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 	drop_nade(get_turf(O))
 
 /datum/ammo/energy/lasgun/marine/heavy_laser/on_hit_turf(turf/T, obj/projectile/P)
-	drop_nade(T.density ? P.loc : T)
+	drop_nade(T.density ? get_step_towards(T, P) : T)
 
 /datum/ammo/energy/lasgun/marine/heavy_laser/do_at_max_range(turf/T, obj/projectile/P)
-	drop_nade(T.density ? P.loc : T)
+	drop_nade(T.density ? get_step_towards(T, P) : T)
+
+//psyblast// lots of placeholder stuff
```

Comment:
hello placeholder stuff

Reply:
Placeholder stuff can't come to the phone right now, can I take a message?